src/model/WaterHeaterStratified.cpp

```diff
@@ -84,102 +84,102 @@ namespace detail {
-          // }
-
-          // TODO: implement these too...
-          //"Water Heater Off Cycle Parasitic <Fuel Type> Rate",
-          //"Water Heater Off Cycle Parasitic <Fuel Type> Energy",
-          //"Water Heater On Cycle Parasitic <Fuel Type> Rate",
-          // "Water Heater On Cycle Parasitic <Fuel Type> Energy",
-          //
-          "Water Heater Water Volume Flow Rate",
-          "Water Heater Water Volume",
-
-
-          // Specific to Stratified
-          "Water Heater Heater 1 Heating Rate",
-          "Water Heater Heater 2 Heating Rate",
-          "Water Heater Heater 1 Heating Energy",
-          "Water Heater Heater 2 Heating Energy",
-          "Water Heater Heater 1 Cycle On Count",
-          "Water Heater Heater 2 Cycle On Count",
-          "Water Heater Heater 1 Runtime Fraction",
-          "Water Heater Heater 2 Runtime Fraction"
+        "Water Heater Final Tank Temperature",
+        "Water Heater Heat Loss Rate",
+        "Water Heater Heat Loss Energy",
+        "Water Heater Use Side Mass Flow Rate",
+        "Water Heater Use Side Inlet Temperature",
+        "Water Heater Use Side Outlet Temperature",
+        "Water Heater Use Side Heat Transfer Rate",
+        "Water Heater Use Side Heat Transfer Energy",
+        "Water Heater Source Side Mass Flow Rate",
+        "Water Heater Source Side Inlet Temperature",
+        "Water Heater Source Side Outlet Temperature",
+        "Water Heater Source Side Heat Transfer Rate",
+        "Water Heater Source Side Heat Transfer Energy",
+        "Water Heater Off Cycle Parasitic Tank Heat Transfer Rate",
+        "Water Heater Off Cycle Parasitic Tank Heat Transfer Energy",
+        "Water Heater On Cycle Parasitic Tank Heat Transfer Rate",
+        "Water Heater On Cycle Parasitic Tank Heat Transfer Energy",
+        "Water Heater Total Demand Heat Transfer Rate",
+        "Water Heater Total Demand Energy",
+        "Water Heater Heating Rate",
+        "Water Heater Heating Energy",
+        "Water Heater Unmet Demand Heat Transfer Rate",
+        "Water Heater Unmet Demand Heat Transfer Energy",
+        "Water Heater Venting Heat Transfer Rate",
+        "Water Heater Venting Heat Transfer Energy",
+        "Water Heater Net Heat Transfer Rate",
+        "Water Heater Net Heat Transfer Energy",
+        "Water Heater Cycle On Count",
+        "Water Heater Runtime Fraction",
+        "Water Heater Part Load Ratio",
+        "Water Heater Electric Power",
+
+        // This should be based on fuel type
+
+        //"Water Heater <Fuel Type> Rate",
+        //"Water Heater <Fuel Type> Energy",
+        // Fuel type specific
+        // TODO: DLM: the return type of this method needs to change to std::vector<std::string> in ModelObject
+        // until then, make this include all possible outputVariableNames for class regardless of fuelType
+        // std::string fuelType = this->fuelType();
+        // if (fuelType == "Electricity") {
```

Comment:
This is an important comment. Some output variables in E+ are added conditionally based on the object's state.
I'm not sure this is worth penalizing all objects were this isn't the case though.